frame/offences/src/lib.rs

```diff
@@ -117,42 +114,12 @@ decl_module! {
 	pub struct Module<T: Config> for enum Call where origin: T::Origin {
 		fn deposit_event() = default;
 
-		fn on_initialize(now: T::BlockNumber) -> Weight {
-			// only decode storage if we can actually submit anything again.
-			if !T::OnOffenceHandler::can_report() {
-				return 0;
-			}
+		fn on_runtime_upgrade() -> Weight {
+			migration::migrate::<T>()
```

Comment:
Could use a better name, given that the migration is doing something specific e.g. `migrations::remove_deferred_storage`